### PR TITLE
Migrate to GitHub Actions

### LINKED ISSUE #547: Migrate to GitHub actions
We can use Macs and recent emulators. Worked great for Retrofit.

### PR DESCRIPTION
**Pull request by ychescale9**
Resolves #547.

Please let me know if you want to test on more / less API levels.

### CODE REVIEW

**Comment by ychescale9** [comment]
Some tests seem to be [broken / flaky](https://github.com/ychescale9/RxBinding/actions/runs/111183481) on newer API levels.

I changed the workflow to replicate the travis setup - only runs on API 18 with 1 module per job. `rxbinding-recyclerview` is the [only failing job](https://github.com/ychescale9/RxBinding/runs/695602201?check_suite_focus=true) now.

I can look at fixing the tests in a separate PR (if you think it's worth it).

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    needs: test
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
```
Comment (JakeWharton) [inline .github/workflows/build.yml:61]: Hopefully this actually works. We might need to make it 8.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      - uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-${{ hashFiles('**/build.gradle') }}-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
+
+      - name: Run Tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 18
```
Comment (JakeWharton) [inline .github/workflows/build.yml:46]: Eventually we'll want to shard this to include minSdk and the latest. And maybe a random value in-between.

**Review by JakeWharton — APPROVED** [review]

**Comment by JakeWharton** [comment]
Can't be worse than what we have now...

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**